### ISSUE TITLE
Why no torch.no_grad() in test?

### ISSUE THREAD

**Issue by IMasI2Cat** [nathan66666/DVMSR#5]
I don't know why you didn't include this on inference (also inference_mode is neither). Aren't you accumulating gradients which should be affecting the used VRAM? Maybe I am missing something?

**Comment by nathan66666** [comment]
Thank you for your interest in our work. By setting the model to `eval()`  mode and manually controlling the `requires_grad` attribute for each parameter, our code ensures that no gradient computation occurs during testing. Although `torch.no_grad()` is not used, the memory usage and computational efficiency are almost unaffected during testing.

**Comment by IMasI2Cat** [comment]
Thank you, I missed this. For anyone wondering, lines 41 to 44 in test_demo.py
